components/nexus-repository/src/main/java/org/sonatype/nexus/repository/storage/StorageTxImpl.java

```diff
@@ -101,7 +105,9 @@ public StorageTxImpl(final BlobTx blobTx,
     this.assetEntityAdapter = checkNotNull(assetEntityAdapter);
     this.hook = checkNotNull(hook);
 
-    db.begin(TXTYPE.OPTIMISTIC);
+    if (!userManagedDb) {
+      db.begin(TXTYPE.OPTIMISTIC);
```

Comment:
Ack fixed in https://github.com/sonatype/nexus-oss/commit/08b853737084db45647dcf63f5e80219ff8b4813